src/protocolsupportpocketstuff/hacks/teams/TeamsPacketListener.java

```diff
@@ -0,0 +1,215 @@
+package protocolsupportpocketstuff.hacks.teams;
+
+import io.netty.buffer.ByteBuf;
+import net.minecraft.server.v1_12_R1.PacketPlayOutScoreboardTeam;
+import protocolsupport.api.Connection;
+import protocolsupport.protocol.serializer.MiscSerializer;
+import protocolsupport.protocol.serializer.StringSerializer;
+import protocolsupport.protocol.serializer.VarNumberSerializer;
+import protocolsupport.protocol.typeremapper.pe.PEPacketIDs;
+import protocolsupport.protocol.utils.datawatcher.DataWatcherObject;
+import protocolsupport.protocol.utils.datawatcher.objects.DataWatcherObjectString;
+import protocolsupport.utils.CollectionsUtils;
+import protocolsupportpocketstuff.ProtocolSupportPocketStuff;
+import protocolsupportpocketstuff.api.util.PocketCon;
+import protocolsupportpocketstuff.packet.play.EntityDataPacket;
+
+import java.lang.reflect.Field;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.List;
+import java.util.UUID;
+
+public class TeamsPacketListener extends Connection.PacketListener {
+	private ProtocolSupportPocketStuff plugin;
+	private Connection con;
+	private HashMap<String, CachedUser> cachedUsers = new HashMap<String, CachedUser>();
+	private HashMap<String, CachedTeam> cachedTeams = new HashMap<String, CachedTeam>();
+
+	public TeamsPacketListener(ProtocolSupportPocketStuff plugin, Connection con) {
+		this.plugin = plugin;
+		this.con = con;
+	}
+
+	@Override
+	public void onPacketSending(PacketEvent event) {
+		super.onPacketSending(event);
+
+		if (!(event.getPacket() instanceof PacketPlayOutScoreboardTeam))
```

Comment:
This will need to be updated for glowstone. Also, I don't feel much for updating the nms from time to time, but I guess its fine for now.